controllers/workspace/provision/deployment.go

```diff
@@ -140,6 +140,31 @@ func DeleteWorkspaceDeployment(ctx context.Context, workspace *devworkspace.DevW
 	return true, nil
 }
 
+// ScaleDeploymentToZero scales the cluster deployment to zero
+func ScaleDeploymentToZero(ctx context.Context, workspace *devworkspace.DevWorkspace, client runtimeClient.Client) error {
+	defaultRetry := time.Millisecond * 100
+	timeout := time.Second * 60
+	err := wait.Poll(defaultRetry, timeout, func() (done bool, err error) {
```

Comment:
I'm not sure we can `wait` within the reconcile loop. This risks tying up a controller thread for 60 seconds when most reconciles take less than a second.

We should instead give the scale down one shot, and if it fails, requeue a reconcile to try it again.

Reply:
fixed!